src/main/telemetry/ibus_shared.c

```diff
@@ -141,15 +141,10 @@ static uint8_t dispatchMeasurementRequest(ibusAddress_t address) {
     }
 #endif
     if (SENSOR_ADDRESS_TYPE_LOOKUP[address].value == IBUS_MEAS_VALUE_TEMPERATURE) { //BARO_TEMP\GYRO_TEMP
-        if (sensors(SENSOR_BARO)) return sendIbusMeasurement2(address, (uint16_t) ((DEGREES_TO_CENTIDEGREES(getCurrentTemperature()) + 50) / 10  + IBUS_TEMPERATURE_OFFSET)); //int32_t
-        else {
-          /*
-           * There is no temperature data
-           * assuming ((DEGREES_TO_CENTIDEGREES(getCurrentTemperature()) + 50) / 10
-           * 0 degrees (no sensor) equals 50 / 10 = 5
-           */
-          return sendIbusMeasurement2(address, (uint16_t) (5 + IBUS_TEMPERATURE_OFFSET)); //int16_t
-        }
+        int16_t temperature;
+        const bool temp_valid = sensors(SENSOR_BARO) ? getBaroTemperature(&temperature) : getIMUTemperature(&temperature);
+        if (!temp_valid || (temperature < -50)) temperature = -50; // Minimum reported temperature is -5°C
+        return sendIbusMeasurement2(address, (uint16_t)((temperature + 50) / 10  + IBUS_TEMPERATURE_OFFSET));
```

Comment:
if getBaroTemperature and getIMUTemperature return temperature in decidegree (0.1) no need to be calculated
Just (uint16_t)(temperature + IBUS_TEMPERATURE_OFFSET)

Reply:
Are you sure about that ?

Reply:
Yes I sure.

Reply:
It doesn't match the old code:

```
return sendIbusMeasurement2(address, (uint16_t) ((DEGREES_TO_CENTIDEGREES(getCurrentTemperature()) + 50) / 10  + IBUS_TEMPERATURE_OFFSET));
```